server/src/logic/services/auth.service.ts

```diff
@@ -3,16 +3,26 @@ import { UserService } from "./user.service";
 import { CryptoUtils } from "./../utils/cryptoUtils";
 import { ErrorUtils } from "./../utils/errorUtils";
 import { Config } from "./../../config";
-import { ApiError, DbError, SocoboUser } from "./../../models/index";
+import { 
+  ApiError, DbError, SocoboUser, LoginResult 
+} from "./../../models/index";
 
 
 export class AuthService {
   constructor (private _userService: UserService) {}
 
-  login (email: string, password: string): Promise<Object> {
+  login (isEmailLogin: boolean, usernameOrEmail: string, password: string): Promise<LoginResult> {
```

Comment:
I would use two methods (loginInWithUsername and logInWithEmail) instead of the flag for the sake of readability but this would maybe blow the code up.

Or you could maybe use two parameters (username and email) and check each only if present.